### PR TITLE
upcoming: [M3-7303] - Replace Managed summary charts with Recharts

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
Replace Managed summary charts with Recharts
- CPU Usage chart
- Network Transfer chart
- Disk I/O chart

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115299789/9b142e0c-d573-4b25-b519-53b462b6f0c8" /> | <video src="https://github.com/linode/manager/assets/115299789/db53e1d0-e878-4393-91b8-932a6dabcf25" /> |

### Prerequisites
(How to setup test environment)
- Have an account that's Managed (you can do so via admin)

### Verification steps 
(How to verify changes)
- Go to `/managed/summary` and verify the updated UI and that there are no regressions in the graph, units, etc 
  - X and Y axis labels will not match exactly but the data from hovering over the graph should still be the same

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/AreaChart.tsx [hunk 1]
```diff
@@ -46,10 +48,19 @@ const humanizeLargeData = (value: number) => {
...
   const theme = useTheme();
 
+  const [activeSeries, setActiveSeries] = React.useState<Array<string>>([]);
+  const handleLegendClick = (dataKey: string) => {
+    if (activeSeries.includes(dataKey)) {
+      setActiveSeries(activeSeries.filter((el) => el !== dataKey));
+    } else {
+      setActiveSeries((prev) => [...prev, dataKey]);
+    }
+  };
```
Comment (hana-linode) [inline packages/manager/src/components/AreaChart.tsx:62]: Code taken from [Recharts Storybook example](https://master--63da8268a0da9970db6992aa.chromatic.com/?path=/story/examples-linechart--hide-on-legend-click)

File: packages/manager/src/features/Managed/ManagedDashboardCard/ManagedChartPanel.tsx [hunk 2]
```diff
@@ -26,16 +30,27 @@ import {
 
 const chartHeight = 300;
 
+interface NetworkTransferProps {
+  'Network Traffic In': number;
+  'Network Traffic Out': number;
+  t: number;
```
Comment (abailly-akamai) [inline packages/manager/src/features/Managed/ManagedDashboardCard/ManagedChartPanel.tsx:36]: Let's use explicit naming conventions for interfaces

**Review by abailly-akamai — COMMENTED** [review]
Looks good and graphs are rendering well. Couple questions:

- Did we get confirmation by UX about legend changes? (ex: standard time VS military time, colors
- As we're implementing AreaChart through the codebase, can we please get it tested and in storybook?
- Is there any plans to get 1/1 parity with legacy charts? Right now this implementation breaks the implementation of the accessible data table

**Review by bnussman-akamai — APPROVED** [review]
Optional, but it would be nice to restore the width (or padding?) here. I think this is a regression from MUI related work. If you want me to create a separate ticket, let me know and I will!

## Expected
Screenshot 2024-01-02 at 2 57 30 PM

## Current
Screenshot 2024-01-02 at 2 40 34 PM

File: packages/manager/src/components/AreaChart/AccessibleAreaChart.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      data &&
+      data.map((entry: any, idx: number) => {
+        const { [dataKey]: value, timestamp } = entry;
+
+        return (
+          <tr key={`accessible-graph-data-body-row-${idx}`}>
+            <td>
+              {timestamp
+                ? DateTime.fromMillis(Number(timestamp)).toLocaleString(
+                    DateTime.DATETIME_SHORT
```
Comment (abailly-akamai) [inline packages/manager/src/components/AreaChart/AccessibleAreaChart.tsx:40]: We should perhaps consider making this a testable chart util as it is likely to be reused in other chart instances

File: packages/manager/src/components/AreaChart/AreaChart.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { AreaChart } from './AreaChart';
+import { timeData } from './utils';
+
+import type { Meta, StoryObj } from '@storybook/react';
+
+const meta: Meta<typeof AreaChart> = {
+  component: AreaChart,
+  title: 'Components/AreaChart',
```
Comment (abailly-akamai) [inline packages/manager/src/components/AreaChart/AreaChart.stories.tsx:10]:
```suggestion
  title: 'Components/Graphs/AreaChart',
```

File: packages/manager/src/components/AreaChart/AreaChart.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    return DateTime.fromMillis(timestamp, { zone: timezone }).toFormat(
+      xAxis.tickFormat
+    );
+  };
+
+  const tooltipLabelFormatter = (timestamp: number) => {
+    return DateTime.fromMillis(timestamp, { zone: timezone }).toFormat(
+      'LLL dd, yyyy, h:mm a'
+    );
+  };
```
Comment (abailly-akamai) [inline packages/manager/src/components/AreaChart/AreaChart.tsx:85]: Same comment regarding making a util

**Review by abailly-akamai — APPROVED** [review]
Thanks for addressing the feedback, and adding accessible table back with tests and stories 🙏  - looks really good ✅ 

Approving pending UX confirmation about treatment and military times. 

- Agree with @bnussman-akamai it would be nice to fix the layout padding or make a ticket for it
- We could also maybe improve the UX from the previous charts by adding either a hover state on the legend or (better) having improved semantic tags
Screenshot 2024-01-03 at 09 24 55
Screenshot 2024-01-03 at 09 24 51

**Comment by hana-linode** [comment]
Confirmed with Andrew that the change from military to standard time is fine. We are going to wait to make any color adjustments until the commencement of the Akamai Component Design System.

**Pull request merged**